pkg/options/types_test.go

```diff
@@ -162,37 +162,38 @@ func TestNodeFieldSelector(t *testing.T) {
 		Wanted string
 	}{
 		{
-			Desc:   "with node name",
+			Desc:   "empty node name",
+			Node:   "",
 			Wanted: "",
 		},
 		{
 			Desc:   "with node name",
-			Node:   nil,
-			Wanted: "",
-		},
-		{
-			Desc: "empty node name",
-			Node: NodeType(
-				map[string]struct{}{
-					"": {},
-				},
-			),
-			Wanted: "spec.nodeName=",
-		},
-		{
-			Desc: "with node name",
-			Node: NodeType(
-				map[string]struct{}{
-					"k8s-node-1": {},
-				},
-			),
+			Node:   "k8s-node-1",
 			Wanted: "spec.nodeName=k8s-node-1",
 		},
 	}
 
 	for _, test := range tests {
 		node := test.Node
-		actual := node.GetNodeFieldSelector()
+		actual := node.GetNodeFieldSelector(false)
+		if !reflect.DeepEqual(actual, test.Wanted) {
+			t.Errorf("Test error for Desc: %s. Want: %+v. Got: %+v.", test.Desc, test.Wanted, actual)
+		}
+	}
+	tests1 := []struct {
+		Desc   string
+		Node   NodeType
+		Wanted string
+	}{
+		{
+			Desc:   "empty node name",
```

Comment:
Can we add a few more tests here? What happens if I set --node and the --fetch-unscheduled-pods argument at the same time? Should it warn about this?